WIP: LeanCode lint (#35)
leancodepl/flutter_corelibrary: packages/leancode_lint/README.md

# leancode_lint

[![leancode_lint pub.dev badge][pub-badge]][pub-badge-link]

Lint rules used internally in LeanCode projects.

## Usage

> Add `leancode_lint` as normal dependency to access `unawaited` function.

Reviewer: ```suggestion
Add `leancode_lint` as a normal dependency to access the `unawaited` function.
```

Reviewer: Also, it's worth mentioning that since Dart 2.14 `unawaited` is available in `dart:async`: https://api.dart.dev/stable/2.14.2/dart-async/unawaited.html

```yaml
dependencies:
  leancode_lint: ^1.0.0
```

### Apps

> Include rules in `analysis_options.yaml` of your project.

Reviewer: ```suggestion
`include: package:leancode_lint/analysis_options.yaml` in `analysis_options.yaml` in your project.
```

If necessary you can exclude some files from analysis.

```dart
include: package:leancode_lint/analysis_options.yaml

# Optional
analyzer:
  exclude:

>     - "**/*.g.dart"

Reviewer: Can we use single quotes instead? Here and @ 40:

```

### Packages

> For packages include `analysis_options_package.yaml` instead.

Reviewer: ```suggestion
For packages include the `analysis_options_package.yaml`.
```

_instead_ refers to something from the previous same-level section. You can't expect the reader to read all sections, only those that they're interested in.
